File: SmartPhone++/SmartPhone.Model/Requests/CartUpsertRequest.cs

using System;
using System.ComponentModel.DataAnnotations;

namespace SmartPhone.Model.Requests
{
    public class CartUpsertRequest
    {
        public DateTime? ExpiresAt { get; set; }
        public bool IsActive { get; set; } = true;
        public int UserId { get; set; }
    }
} 

File: SmartPhone++/UI/smartphone_mobile_client/lib/model/cart_upsert_request.dart
import 'package:json_annotation/json_annotation.dart';

part 'cart_upsert_request.g.dart';

@JsonSerializable()
class CartUpsertRequest {
  final DateTime? expiresAt;
  final bool isActive;
  final int userId;

  CartUpsertRequest({
    this.expiresAt,
    this.isActive = true,
    required this.userId,
  });

  factory CartUpsertRequest.fromJson(Map<String, dynamic> json) => _$CartUpsertRequestFromJson(json);
  Map<String, dynamic> toJson() => _$CartUpsertRequestToJson(this);
}
